sklearn/decomposition/_kernel_pca.py

```diff
@@ -215,6 +268,11 @@ def _fit_transform(self, K):
                                                 tol=self.tol,
                                                 maxiter=self.max_iter,
                                                 v0=v0)
+        elif eigen_solver == 'randomized':
+            self.lambdas_, self.alphas_ = _randomized_eigsh(
+                K, n_components=n_components, n_iter=self.iterated_power,
+                random_state=self.random_state, selection='module'
+            )
```

Comment:
There should be an `else:` clause to raise `ValueError` with an informative error message that recalls the valid choices for the `eigen_solver` parameter along with the current value of `eigen_solver`. At the moment we have the following cryptic error message:

```python-traceback
>>> import numpy as np
>>> from sklearn.decomposition import KernelPCA
>>> KernelPCA(eigen_solver="invalid").fit(np.random.randn(10, 10))
Traceback (most recent call last):
  File "<ipython-input-7-3be47a964842>", line 1, in <module>
    KernelPCA(eigen_solver="invalid").fit(np.random.randn(10, 10))
  File "/Users/ogrisel/code/scikit-learn/sklearn/decomposition/_kernel_pca.py", line 344, in fit
    self._fit_transform(K)
  File "/Users/ogrisel/code/scikit-learn/sklearn/decomposition/_kernel_pca.py", line 278, in _fit_transform
    self.lambdas_ = _check_psd_eigenvalues(self.lambdas_,
AttributeError: 'KernelPCA' object has no attribute 'lambdas_'
```

Also please add a test.

Reply:
I'm so happy that you suggest this ! Once again this is the kind of things that I was reluctant to change as this was how sklearn was made :)